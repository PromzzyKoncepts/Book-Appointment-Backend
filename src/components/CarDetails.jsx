import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { fetchCar, car } from '../redux/cars/carsSlice';

const CarDetails = () => {
  const dispatch = useDispatch();
  const id = useParams();
  const carId = Number(id.id);
  const cardetails = useSelector(car);
  useEffect(() => {
    dispatch(fetchCar(carId));
  }, [dispatch]);

  return (
    <>
      <div className="car-container" key={cardetails.id}>
        <img src={cardetails.image_url} alt={cardetails.name} />
        <div className="car_details">
          <div className="car-header">
            <h2>{cardetails.name}</h2>
            <p>{cardetails.description}</p>
          </div>
          <div className="space-between fee">
            <p>Finance Fee</p>
            <p>$139</p>
          </div>
          <div className="space-between">
            <p>Option to purchase Fee</p>
            <p>$249</p>
          </div>
          <div className="space-between fee">
            <p>Total amount payable</p>
            <p>{cardetails.price}</p>
          </div>
          <div className="space-between">
            <p>Duration</p>
            <p>48 Months</p>
          </div>
          <div className="space-between">
            <p>Availability:</p>
            <p>
              {cardetails.reserved === false ? (
                <span>Not available</span>
              ) : (
                <span>available</span>
              )}
            </p>
          </div>
          <div className="more-model">
            <span className="desc_short">We can satisfy your desire!</span>
          </div>
          <Link to="/reservations" className="btn">
            <button
              type="button"
              className="reserve"
            >
              Reserve
            </button>
          </Link>
        </div>
      </div>
    </>
  );
};

export default CarDetails;
