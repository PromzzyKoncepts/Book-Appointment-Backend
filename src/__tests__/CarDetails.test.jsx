import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/';
import CarDetails from '../components/CarDetails';

describe('CarDetails component', () => {
  test('Should render the proper CarDetails component', () => {
    const carDetailscomponent = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <CarDetails />
          </Router>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(carDetailscomponent).toMatchSnapshot();
  });
});