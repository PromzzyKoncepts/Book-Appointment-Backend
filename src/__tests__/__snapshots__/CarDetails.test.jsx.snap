// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetails component Should render the proper CarDetails component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="car-container"
      >
        <img />
        <div
          class="car_details"
        >
          <div
            class="car-header"
          >
            <h2 />
            <p />
          </div>
          <div
            class="space-between fee"
          >
            <p>
              Finance Fee
            </p>
            <p>
              $139
            </p>
          </div>
          <div
            class="space-between"
          >
            <p>
              Option to purchase Fee
            </p>
            <p>
              $249
            </p>
          </div>
          <div
            class="space-between fee"
          >
            <p>
              Total amount payable
            </p>
            <p />
          </div>
          <div
            class="space-between"
          >
            <p>
              Duration
            </p>
            <p>
              48 Months
            </p>
          </div>
          <div
            class="space-between"
          >
            <p>
              Availability:
            </p>
            <p>
              <span>
                available
              </span>
            </p>
          </div>
          <div
            class="more-model"
          >
            <span
              class="desc_short"
            >
              We can satisfy your desire!
            </span>
          </div>
          <a
            class="btn"
            href="/book_car"
          >
            <button
              class="reserve"
              type="button"
            >
              Reserve
            </button>
          </a>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="car-container"
    >
      <img />
      <div
        class="car_details"
      >
        <div
          class="car-header"
        >
          <h2 />
          <p />
        </div>
        <div
          class="space-between fee"
        >
          <p>
            Finance Fee
          </p>
          <p>
            $139
          </p>
        </div>
        <div
          class="space-between"
        >
          <p>
            Option to purchase Fee
          </p>
          <p>
            $249
          </p>
        </div>
        <div
          class="space-between fee"
        >
          <p>
            Total amount payable
          </p>
          <p />
        </div>
        <div
          class="space-between"
        >
          <p>
            Duration
          </p>
          <p>
            48 Months
          </p>
        </div>
        <div
          class="space-between"
        >
          <p>
            Availability:
          </p>
          <p>
            <span>
              available
            </span>
          </p>
        </div>
        <div
          class="more-model"
        >
          <span
            class="desc_short"
          >
            We can satisfy your desire!
          </span>
        </div>
        <a
          class="btn"
          href="/book_car"
        >
          <button
            class="reserve"
            type="button"
          >
            Reserve
          </button>
        </a>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
